service/operator/service_test.go

```diff
@@ -1,95 +1,604 @@
 package operator
 
 import (
+	"reflect"
 	"testing"
 
+	"github.com/giantswarm/ingresstpr"
+	"github.com/giantswarm/ingresstpr/guestcluster"
+	"github.com/giantswarm/ingresstpr/hostcluster"
+	"github.com/giantswarm/ingresstpr/hostcluster/ingresscontroller"
+	"github.com/giantswarm/ingresstpr/protocolport"
 	apiv1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/util/intstr"
 )
 
-func Test_Service_newClusterPort(t *testing.T) {
+func Test_Service_GetDesiredState(t *testing.T) {
 	testCases := []struct {
-		K8sPorts       []apiv1.ServicePort
-		AvailablePorts []int
-		Expected       int
-		ErrorMatcher   func(error) bool
+		Obj          interface{}
+		Expected     DesiredState
+		ErrorMatcher func(error) bool
```

Comment:
The tests are nice and they show that the `Operator` interface is also nice as well. All state transforming methods are freed from external dependencies. They implement only data processing algorithms without the need of clients connecting to remote service via network. The tests show pretty good how certain state is transformed.